recipes/symengine/all/conanfile.py

```diff
@@ -22,7 +22,7 @@ class SymengineConan(ConanFile):
     default_options = {
         "shared": False,
         "fPIC": True,
-        "integer_class": "boostmp",
+        "integer_class": "gmp",
```

Comment:
It affects directly the default package ID. Why is needed to change it now?

Reply:
@cqc-alec Keep `gmp` please. After checking the upstream, that's the default option value. Thank you for fixing it. Reference: https://github.com/symengine/symengine/blob/v0.9.0/CMakeLists.txt#L145

Reply:
yes, it's better if our defaults match upstream defaults, it would meet user's expectations